collectors/30/kafka_offset_collector.py

```diff
@@ -66,26 +88,33 @@ def get_partitions(topic):
 def report():
     # see the monitoring.json chef role
     zk_quorums = os.getenv('MONITORED_ZOOKEEPER_QUORUMS')
-    kafka_brokers = os.getenv('KAFKA_BROKERS')
 
     if zk_quorums is None:
         raise RuntimeError('MONITORED_ZOOKEEPER_QUORUMS not found')
 
-    if kafka_brokers is None:
-        raise RuntimeError('KAFKA_BROKERS not found')
-
-    # we could just use ZK to find all topics, should we just do that?
-    topics = ['mobile_metrics', 'raw_events']
-
     consumer_group = 'secor_group'
 
     for zk_quorum in zk_quorums.split('|'):
         zk = KazooClient(hosts=zk_quorum)
-        kafka = KafkaClient(kafka_brokers)
-
         zk.start()
+
+        kafka_brokers = get_kafka_brokers(zk)
+        if len(kafka_brokers) is 0:
+            raise RuntimeError('KAFKA_BROKERS could not be fetched from ZK')
+        kafka_init = 'env=production kafka=%s zk=%s' % (','.join(kafka_brokers), zk_quorum)
+        kafka = KafkaClient(kafka_init)
+
+        # Pull topics from zookeeper
+        # topics = ['mobile_metrics', 'raw_events']
+        topics = get_kafka_topics(zk)
+        if len(topics) is 0:
+            raise RuntimeError('KAFKA_TOPICS could not be fetched from ZK')
+        topics = map(str, topics)
         for topic in topics:
-            total_broker_offset = report_broker_info(kafka, zk, topic)
+            try:
+                total_broker_offset = report_broker_info(kafka, zk, topic)
+            except:
```

Comment:
Similarly log the error here.
```py
except Exception as err:
  logging.exception(err)
  # fallback code
```
Also it's considered good practice not to use a bare `except` or `except Exception` but actually anticipate what kind of `Exception`(s) can be thrown and catch those. Python is not your friend though in helping you figure out what subclasses of `Exception` can happen (you have to hope the client code is well documented in that regard).